spec/services/charts/home/categories_with_changes_date_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+describe Charts::Home::CategoriesWithChangesDate do
+  let(:user) { create(:user) }
+
+  let(:balance) { Faker::Number.decimal(4, 2).to_f }
+
+  let(:category) { create(:category, user: user, categorizable: user) }
+  let(:amount) { Faker::Number.decimal(2, 2).to_f }
+
+  let(:categories_titles) { categories.pluck(:title) }
+
+  let(:changes) do
+    Charts::Home::CategoriesWithChanges.new(Transaction.category_transactions).call
+  end
+
+  let(:data_chenges) { described_class.new(changes).date }
+
+  before do
+    create(:balance_transaction_creator, amount: balance, user: user)
+
+    5.times do
+      CategoryTransactions::Creator.new(
+        amount: amount,
+        user_id: user.id,
+        category_id: category.id
+      ).create
+    end
```

Comment:
@marderer let's go with similar solution with `factory :balance_transaction_creator` and use `create_list`

Reply:
@AleksSenkou I tried but I didn't succeed. I spent more than a day and did not work, can I later ask you about this and fix it in another PR?

Reply:
@marderer yeah, we can discuss it later